src/lint.cpp

```diff
@@ -642,6 +641,12 @@ void linter::report_error_if_assignment_is_illegal(
       }
     }
     break;
+  case variable_kind::_import:
+    if (!is_global_variable) {
```

Comment:
Nit: I think this check is unnecessary. `is_global_variable` should always be `false` for `_import` variables. Maybe we could assert:
```suggestion
    QLJS_ASSERT(!is_global_variable);
```

Reply:
I didn't know QLJS_ASSERT worked like a if release mode 😬

Reply:
Actually, this didn't work. I wrote the `if` explicitly because "GCC Release 8" was failing on dereferencing null pointer in line 646 of lint.cpp, even though we check in line 625. Using QLJS_ASSERT, as you suggested, didn't fix this issue. Maybe the explicit `if` is necessary here?

Reply:
> Actually, this didn't work. I wrote the `if` explicitly because "GCC Release 8" was failing on dereferencing null pointer in line 646 of lint.cpp, even though we check in line 625.

Sounds like a false positive compiler warning. We can disable the warning. See this for example: https://github.com/quick-lint/quick-lint-js/blob/9463bbe21754fb37ecb997ecea7c1375fdd912b8/src/lint.cpp#L652-L659

I'm fine with the `if` style.